src/extension/app-schema/app-schema-test/src/test/java/org/geoserver/test/NamespacesWfsTest.java

```diff
@@ -336,4 +346,80 @@ private void checkCount(
             throw new RuntimeException("Error evaluating xpath.", exception);
         }
     }
+
+    /**
+     * Test a request with two queries (to different featureTypes and namespaces) Checks null
+     * prefixes issue on multiple query WFS 2.0.0 GML 3.2 versions
+     */
+    @Test
+    public void testTwoQueriesNamespacesGml32() throws Exception {
+        String wfsQuery =
+                IOUtils.toString(
+                        getClass()
+                                .getClassLoader()
+                                .getResourceAsStream(
+                                        "test-data/stations/stations_two_queries.xml"));
+        Document document = postAsDOM("wfs", wfsQuery);
+        checkCount(
+                WFS20_XPATH_ENGINE,
+                document,
+                1,
+                "//wfs:FeatureCollection/wfs:member/wfs:FeatureCollection/wfs:member/st_gml32:Station_gml32");
+
+        checkCount(
+                WFS20_XPATH_ENGINE,
+                document,
+                1,
+                "/wfs:FeatureCollection/wfs:member/wfs:FeatureCollection/wfs:member/"
+                        + "ms_gml32:Measurement_gml32");
+        // check prefixes:
+        String output = toString(document);
+        assertTrue(output.indexOf("null:Measurement_gml32") < 0);
+        assertTrue(output.indexOf("null:Station_gml32") < 0);
+        assertTrue(output.indexOf("ms_gml32:Measurement_gml32") > -1);
```

Comment:
Why -1 ? It should always be be greater than zero, no ?

Reply:
String.indexOf returns the index of the first occurrence of the specified substring, or -1 if there is no such occurrence.  So if is == -1 substring not found.

Reply:
Ah! right my bad, I thought you were doing a XPATH count.